src/index.ts

```diff
@@ -7,71 +9,77 @@ async function retry<T>(f: () => Promise<T>): Promise<T> {
       finalError = e;
       if (i <= 5) {
         const timeout = Math.min(500, 100 * 2 ** i);
-        await new Promise((r) => setTimeout(r, timeout));
+        await delay(timeout);
       } else {
-        console.warn(`error ${e} retrying ${5 - i} more times.`);
+        debug(`error ${e} retrying ${5 - i} more times.`);
         break;
       }
     }
   }
   throw finalError;
 }
 
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
 export type JsonValue = ReturnType<typeof JSON.parse>;
 export type Formatter<T> = (row: JsonValue[]) => T;
 type DefaultType = { [key: string]: JsonValue };
 
-/**
- * Represents a request to the API
- * @template T The expected return type of the formatted data
- * @see {@link https://www.indexsupply.net/docs#get-query GET /query API documentation}
- * @see {@link https://www.indexsupply.net/docs#chains Supported chains}
- * @see {@link https://www.indexsupply.net/docs#sql SQL query syntax}
- * @example
- * interface Transfer { from: string; to: string; value: string }
- *
- * const request: Request<Transfer> = {
- *   chainId: 1n,
- *   query: 'SELECT "from", "to", value FROM transfer',
- *   eventSignatures: ["Transfer(address indexed from, address indexed to, uint256 value)"],
- *   formatRow: (row) => ({
- *     from: row[0] as string,
- *     to: row[1] as string,
- *     value: row[2] as string
- *   })
- * }
- */
-export type Request<T> = {
-  /** Optional custom API URL. Defaults to https://api.indexsupply.net */
-  apiUrl?: string;
-  /** Optional API key for authentication. Unauthenticated requests limited to 5 per minute */
-  apiKey?: string;
-  /** Chain ID for the target blockchain */
-  chainId: bigint;
-  /** SQL query to execute */
-  query: string;
-  /** Optional array of event signatures to filter events */
-  eventSignatures?: ReadonlyArray<string>;
-  /** Optional function to format the row data. Required if T is not DefaultType */
-  formatRow?: T extends DefaultType ? undefined | Formatter<T> : Formatter<T>;
-};
+namespace IS {
```

Comment:
@ryandotsmith I'm curious what you think about a namespace for the Request and Response types. We will run into type issues because `fetch` returns a `Response` type that is built-in, and we are overriding that type with our own `Response` type.

Reply:
neat. i didn't realize `namespace` existed

Reply:
Does this impact callers?

Reply:
I'm not 100% what you mean, but if someone wanted to use the type they would need to include the namespace.